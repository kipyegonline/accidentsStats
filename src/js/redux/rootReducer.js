const initState = {
  ntsa: [],
  ntsb: [],
  loadedA: false,
  loadedB: false,
  iclass: [],
  auth: { isLoggedIn: true, userName: "kipyegonline" },
};

function rootReducer(state = initState, action) {
  if (action.type === "ADD_NTSA") {
    return { ...state, ntsa: action.payload, loadedA: true };
  } else if (action.type === "ADD_NTSB") {
    return { ...state, ntsb: action.payload, loadedB: true };
  } else if (action.type === "ADD_CLASS") {
    return { ...state, iclass: action.payload };
  } else {
    return state;
  }
}
export default rootReducer;
