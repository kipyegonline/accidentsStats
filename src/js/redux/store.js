import { createStore } from "redux";
import rootReducer from "./rootReducer";
const devtools = window.__REDUX_DEVTOOLS_EXTENSION__
  ? window.__REDUX_DEVTOOLS_EXTENSION__()
  : (f) => f;
export const store = createStore(rootReducer, devtools);
