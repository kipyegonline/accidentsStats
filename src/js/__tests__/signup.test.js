import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";
import SignUp, { DataInput } from "../components/auth/signup";
import Login from "../components/auth/login";
import toJson from "enzyme-to-json";
import { MockStore } from "./redux.test";
import * as $ from "jquery";

Enzyme.configure({ adapter: new Adapter() });
//mounting the sign up form
describe("Mount the form", () => {
  const wrapper = mount(
    <MockStore>
      <SignUp />
    </MockStore>
  );
  it("matches snapshot", () => {
    expect(toJson(wrapper.find("form"))).toMatchSnapshot();
  });
  it("has 5 input elements", () => {
    expect(wrapper.find("input")).toHaveLength(5);
  });
  it("has 3 p tags", () => expect(wrapper.find("p")).toHaveLength(2));
  it("submits", () => {
    wrapper.find("form").simulate("submit");

    //I need to find out how to set state on react hooks
    expect(wrapper.find("p.text-danger").text()).toBe("Name cannot be blank");
  });
});
//shallow render the inputs of the form
describe("<LoginInput />", () => {
  const _sendValue = jest.fn();
  const wrapper = shallow(
    <DataInput
      sendValue={_sendValue}
      id="email"
      type="email"
      name="email"
      classlist="form-control"
      placeholder="Enter email"
    />
  );
  //skipping since that blur event is not passed by prop
  test.skip("Props", () => {
    expect(wrapper.props()).toEqual({
      id: "email",
      type: "email",
      onBlur: (f) => f,
      name: "email",
      placeholder: "Enter email",
      className: "form-control mb-2",
    });
  });
  test("Change event", () => {
    wrapper.simulate("change", {
      target: {
        value: "The email",
        name: "email",
      },
    });
    expect(_sendValue).toBeCalled();
    expect(_sendValue).toBeCalledWith("The email");
    expect(_sendValue).toHaveReturnedWith(undefined);
  });
}); //describe

//The login form, e=requires mounting since i have tested the input components
describe("<Login/>", () => {
  const wrapper = mount(
    <MockStore>
      <Login />
    </MockStore>
  );
  //wrapper.find("form").instance().email = "vinnykipx@.com";
  it("Matches snapshot", () =>
    expect(toJson(wrapper.find("form"))).toMatchSnapshot());
  it("has 3 input elements", () =>
    expect(wrapper.find("input")).toHaveLength(3));
  it("has 2 p tags", () => expect(wrapper.find("p")).toHaveLength(2));
  it("doesn't submit", () => {
    wrapper.find("form").simulate("submit");
    expect(
      wrapper
        .find("p")
        .at(0)
        .text()
    ).toBe("Enter an email address.");
  });
});
