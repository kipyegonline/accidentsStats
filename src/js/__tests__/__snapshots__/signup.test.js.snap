// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> Matches snapshot 1`] = `
<form
  className="form p-3 mx-auto"
  onSubmit={[Function]}
>
  <h3
    className="text-center p-2"
  >
    Login
  </h3>
  <FormGroup
    tag="div"
  >
    <div
      className="form-group"
    >
      <Label
        className="mb-2"
        htmlFor="login-email"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <label
          className="mb-2"
          htmlFor="login-email"
        >
          Email
        </label>
      </Label>
      <DataInput
        classlist="form-ccontrol"
        id="login-email"
        name="email"
        placeholder=""
        sendValue={[Function]}
        type="email"
      >
        <Input
          className="form-ccontrol"
          id="login-email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          type="email"
        >
          <input
            className="form-ccontrol form-control"
            id="login-email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="email"
          />
        </Input>
      </DataInput>
    </div>
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <div
      className="form-group"
    >
      <Label
        className="mb-2"
        htmlFor="login-password"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <label
          className="mb-2"
          htmlFor="login-password"
        >
          Password
        </label>
      </Label>
      <DataInput
        classlist="form-ccontrol"
        id="login-password"
        name="set-password"
        placeholder=""
        sendValue={[Function]}
        type="password"
      >
        <Input
          className="form-ccontrol"
          id="login-password"
          name="set-password"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          type="password"
        >
          <input
            className="form-ccontrol form-control"
            id="login-password"
            name="set-password"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="password"
          />
        </Input>
      </DataInput>
    </div>
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <div
      className="form-group"
    >
      <p
        className="text-danger"
      />
      <p
        className="text-success"
      />
    </div>
  </FormGroup>
  <input
    className="btn btn-primary btn-block"
    id="login-submit"
    name="submit"
    placeholder=""
    type="submit"
    value="Login"
  />
</form>
`;

exports[`Mount the form matches snapshot 1`] = `
<form
  className="form"
  onSubmit={[Function]}
  style={
    Object {
      "background": "#fefefe",
      "margin": "1rem auto",
      "maxWidth": 600,
      "padding": "1rem",
    }
  }
>
  <h5
    className="text-center p-1 my-2"
  >
    Sign up section.
  </h5>
  <FormGroup
    tag="div"
  >
    <div
      className="form-group"
    >
      <Label
        htmlFor="username"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <label
          className=""
          htmlFor="username"
        >
          Username
        </label>
      </Label>
      <DataInput
        classlist="form-control mb-2"
        id="username"
        name="username"
        placeholder=""
        sendValue={[Function]}
        type="text"
      >
        <Input
          className="form-control mb-2"
          id="username"
          name="username"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          type="text"
        >
          <input
            className="form-control mb-2 form-control"
            id="username"
            name="username"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="text"
          />
        </Input>
      </DataInput>
    </div>
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <div
      className="form-group"
    >
      <Label
        htmlFor="email"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <label
          className=""
          htmlFor="email"
        >
          Email
        </label>
      </Label>
      <DataInput
        classlist="form-control mb-2"
        id="email"
        name="email"
        placeholder=""
        sendValue={[Function]}
        type="email"
      >
        <Input
          className="form-control mb-2"
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          type="email"
        >
          <input
            className="form-control mb-2 form-control"
            id="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="email"
          />
        </Input>
      </DataInput>
    </div>
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <div
      className="form-group"
    >
      <Label
        htmlFor="password"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <label
          className=""
          htmlFor="password"
        >
          Password
        </label>
      </Label>
      <DataInput
        classlist="form-control mb-2"
        id="password"
        name="password"
        placeholder=""
        sendValue={[Function]}
        type="password"
      >
        <Input
          className="form-control mb-2"
          id="password"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          type="password"
        >
          <input
            className="form-control mb-2 form-control"
            id="password"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="password"
          />
        </Input>
      </DataInput>
    </div>
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <div
      className="form-group"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <label
          className=""
        >
          Confirm Passsword
        </label>
      </Label>
      <DataInput
        classlist="form-control mb-2"
        id="confirmPassword"
        name="confirmPassword"
        placeholder=""
        sendValue={[Function]}
        type="password"
      >
        <Input
          className="form-control mb-2"
          id="confirmPassword"
          name="confirmPassword"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          type="password"
        >
          <input
            className="form-control mb-2 form-control"
            id="confirmPassword"
            name="confirmPassword"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="password"
          />
        </Input>
      </DataInput>
    </div>
  </FormGroup>
  <p
    className="text-danger text-center my-2"
  />
  <p
    className="text-sucess text-center my-2"
  />
  <DataInput
    classlist="btn btn-block indigo text-white"
    id=""
    name=""
    placeholder=""
    sendValue={[Function]}
    type="submit"
  >
    <Input
      className="btn btn-block indigo text-white"
      id=""
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      placeholder=""
      type="submit"
    >
      <input
        className="btn btn-block indigo text-white form-control"
        id=""
        name=""
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        type="submit"
      />
    </Input>
  </DataInput>
</form>
`;
